eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs

```diff
@@ -461,52 +470,60 @@ validateExUnitsTooBigUTxO pp tx =
 --
 -- > ‖collateral tx‖  ≤  maxCollInputs pp
 validateTooManyCollateralInputs ::
-  ( HasField "_maxCollateralInputs" (Core.PParams era) Natural,
-    HasField "collateral" (Core.TxBody era) (Set a)
+  ( HasField "_maxCollateralInputs" (PParams era) Natural,
+    AlonzoEraTxBody era
   ) =>
-  Core.PParams era ->
-  Core.TxBody era ->
+  PParams era ->
+  TxBody era ->
   Test (UtxoPredicateFailure era)
-validateTooManyCollateralInputs pp txb =
+validateTooManyCollateralInputs pp txBody =
   failureUnless (numColl <= maxColl) $ TooManyCollateralInputs maxColl numColl
   where
     maxColl, numColl :: Natural
     maxColl = getField @"_maxCollateralInputs" pp
-    numColl = fromIntegral . Set.size $ getField @"collateral" txb
+    numColl = fromIntegral . Set.size $ txBody ^. collateralInputsTxBodyL
 
 -- ================================================================
 
 -- | The UTxO transition rule for the Alonzo eras.
 utxoTransition ::
   forall era.
-  ( Era era,
-    ValidateScript era,
-    ConcreteAlonzo era, -- Unlike the Tests, we are ony going to use this once, so we fix the Core.XX types
-    Core.Tx era ~ ValidatedTx era,
-    Core.Witnesses era ~ TxWitness era,
+  ( AlonzoEraTx era,
+    Tx era ~ AlonzoTx era,
     STS (AlonzoUTXO era),
     -- instructions for calling UTXOS from AlonzoUTXO
-    Embed (Core.EraRule "UTXOS" era) (AlonzoUTXO era),
-    Environment (Core.EraRule "UTXOS" era) ~ Shelley.UtxoEnv era,
-    State (Core.EraRule "UTXOS" era) ~ Shelley.UTxOState era,
-    Signal (Core.EraRule "UTXOS" era) ~ Core.Tx era,
-    Inject (PredicateFailure (Core.EraRule "PPUP" era)) (PredicateFailure (Core.EraRule "UTXOS" era))
+    Embed (EraRule "UTXOS" era) (AlonzoUTXO era),
+    Environment (EraRule "UTXOS" era) ~ Shelley.UtxoEnv era,
+    State (EraRule "UTXOS" era) ~ Shelley.UTxOState era,
+    Signal (EraRule "UTXOS" era) ~ Tx era,
+    HasField "_poolDeposit" (PParams era) Coin,
+    HasField "_minfeeA" (PParams era) Natural,
+    HasField "_minfeeB" (PParams era) Natural,
+    HasField "_keyDeposit" (PParams era) Coin,
+    HasField "_maxValSize" (PParams era) Natural,
+    HasField "_maxTxSize" (PParams era) Natural,
+    HasField "_maxTxExUnits" (PParams era) ExUnits,
+    HasField "_protocolVersion" (PParams era) ProtVer,
+    HasField "_coinsPerUTxOWord" (PParams era) Coin,
+    HasField "_maxCollateralInputs" (PParams era) Natural,
+    HasField "_collateralPercentage" (PParams era) Natural,
```

Comment:
I suppose removing the `HasField`s from the protocol parameters is a good candidate to clean up next?